test/rocprim/test_device_radix_sort.hpp

```diff
@@ -1393,4 +1393,84 @@ void sort_keys_over_4g()
     }
 }
 
+inline void sort_keys_large_sizes()
+{
+    int device_id = test_common_utils::obtain_device_from_ctest();
+    SCOPED_TRACE(testing::Message() << "with device_id= " << device_id);
+    HIP_CHECK(hipSetDevice(device_id));
+
+    using key_type                   = uint8_t;
+    constexpr unsigned int start_bit = 0;
+    constexpr unsigned int end_bit   = 8;
+
+    hipStream_t stream = 0;
+
+    const std::vector<size_t> sizes = test_utils::get_large_sizes(seeds[0]);
```

Comment:
Could we temporarily change this to eliminate the sizes the surpass our CI systems' 96 GB memory allocation limit? Our CI config will be adjusted to allow larger requests in the future (at which point we could revert back to calling test_utils::get_large_sizes), but not in time for the merge deadline for this PR.
```suggestion
    // Currently, CI enforces a hard limit of 96 GB on memory allocations.
    // Instead of calling test_utils::get_large_sizes here, temporarily
    // build a local vector that consists of sizes that will require less
    // space than the limit.
    std::vector<size_t> sizes = {
        (size_t{1} << 30) - 1,
        size_t{1} << 31,
        (size_t{1} << 32) - 15,
        (size_t{1} << 33) + (size_t{1} << 32) - 876543,
        (size_t{1} << 34) - 12346,
        (size_t{1} << 35) + 1,
        (size_t{1} << 36) - 1,
    };
    const std::vector<size_t> random_sizes
        = test_utils::get_random_data<size_t>(2,
                                              (size_t{1} << 30) + 1,
                                              (size_t{1} << 36) - 2,
                                              seeds[0]);
    sizes.insert(sizes.end(), random_sizes.begin(), random_sizes.end());
    std::sort(sizes.begin(), sizes.end());
```

Reply:
Applied a slightly different workaround, just to avoid code duplication. Should be equivalent to the one suggested

Reply:
Thanks, this looks great!